crates/amaru-kernel/src/protocol_parameters.rs

```diff
@@ -7,7 +7,7 @@ pub struct ProtocolParameters {
     pub max_header_size: u32,
     pub max_tx_ex_units: ExUnits,
     pub max_block_ex_units: ExUnits,
-    pub max_val_size: u32,
```

Comment:
`u32` looks like the correct value, why the change?

Reply:
[usize](https://doc.rust-lang.org/std/primitive.usize.html) is platform dependent (32 bits on 32 bits system, 64 on 64, ..) so it doesn't look right here. Given this is something that we get from network and others node (via updates, snapshots, ..), it probably is best to stick with what is described in the [CDDL](https://github.com/IntersectMBO/cardano-ledger/blob/2d1e94cf96d00ba0da53883c388fa0aba6d74624/eras/conway/impl/cddl-files/conway.cddl#L373) (so `u32` in this case?)

I tried to align all values accordingly in this [PR](https://github.com/pragma-org/amaru/pull/204)

Reply:
Fair enough, I will change it back. I don't love having to do type conversions, but I think I agree it makes more sense for the `ProtocolParameters` type to match the CDDL